src/displayapp/screens/Dice.cpp

```diff
@@ -0,0 +1,93 @@
+
+#include "displayapp/screens/Screen.h"
+#include "displayapp/screens/Symbols.h"
+
+#include <lvgl/lvgl.h>
+
+using namespace Pinetime::Applications::Screens;
+
+namespace {
+  lv_obj_t* MakeLabel(lv_font_t* font, lv_color_t color, lv_label_long_mode_t longMode, uint8_t width, lv_label_align_t labelAlignment, const char* text, lv_obj_t* reference, lv_align_t alignment, int8_t x, int8_t y) {
+    lv_obj_t* label = lv_label_create(lv_scr_act(), nullptr);
+    lv_obj_set_style_local_text_font(label, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, font);
+    lv_obj_set_style_local_text_color(label, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, color);
+    lv_label_set_long_mode(label, longMode);
+    if (width != 0) {
+      lv_obj_set_width(label, width);
+    }
+    lv_label_set_align(label, labelAlignment);
+    lv_label_set_text(label, text);
+    lv_obj_align(label, reference, alignment, x, y);
+    return label;
+  }
+
+  static void btnRollEventHandler(lv_obj_t* obj, lv_event_t event) {
+    auto* screen = static_cast<Dice*>(obj->user_data);
+    if (event == LV_EVENT_CLICKED) {
+      screen->Roll();
+    }
+  }
+}
+
+Dice::Dice(DisplayApp* app, Controllers::MotionController& motionController) : Screen(app) {
+  std::seed_seq sseq{xTaskGetTickCount(), static_cast<uint32_t>(motionController.X()), static_cast<uint32_t>(motionController.Y()), static_cast<uint32_t>(motionController.Z())};
+  gen.seed(sseq);
+
+  nCounter.Create();
+  lv_obj_align(nCounter.GetObject(), lv_scr_act(), LV_ALIGN_IN_TOP_LEFT, 0, 24);
+  nCounter.SetValue(1);
+
+  lv_obj_t* dLabel = MakeLabel(&jetbrains_mono_42, LV_COLOR_WHITE, LV_LABEL_LONG_EXPAND, 0, LV_LABEL_ALIGN_LEFT, "d", nCounter.GetObject(), LV_ALIGN_OUT_RIGHT_MID, 0, 0);
+
+  dCounter.Create();
+  lv_obj_align(dCounter.GetObject(), dLabel, LV_ALIGN_OUT_RIGHT_MID, 2, 0);
+  dCounter.SetValue(6);
+
+  std::uniform_int_distribution<> distrib(0, resultColors.size()-1);
+  currentColorIndex = distrib(gen);
+
+  resultTotalLabel = MakeLabel(&jetbrains_mono_42, resultColors[currentColorIndex], LV_LABEL_LONG_BREAK, 120, LV_LABEL_ALIGN_CENTER, "", lv_scr_act(), LV_ALIGN_IN_TOP_RIGHT, 13, 30);
+  resultIndividualLabel = MakeLabel(&jetbrains_mono_bold_20, resultColors[currentColorIndex], LV_LABEL_LONG_BREAK, 90, LV_LABEL_ALIGN_CENTER, "", resultTotalLabel, LV_ALIGN_OUT_BOTTOM_MID, 0, 10);
+
+  Roll();
+
+  btnRoll = lv_btn_create(lv_scr_act(), nullptr);
+  btnRoll->user_data = this;
+  lv_obj_set_event_cb(btnRoll, btnRollEventHandler);
+  lv_obj_set_size(btnRoll, 240, 50);
+  lv_obj_align(btnRoll, lv_scr_act(), LV_ALIGN_IN_BOTTOM_MID, 0, 0);
+
+  btnRollLabel = MakeLabel(&jetbrains_mono_bold_20, LV_COLOR_WHITE, LV_LABEL_LONG_EXPAND, 0, LV_LABEL_ALIGN_CENTER, Symbols::dice, btnRoll, LV_ALIGN_CENTER, 0, 0);
+}
+
+Dice::~Dice() {
+  lv_obj_clean(lv_scr_act());
+}
+
+void Dice::Roll() {
+  uint8_t resultIndividual;
+  uint16_t resultTotal = 0;
+  std::uniform_int_distribution<> distrib(1, dCounter.GetValue());
```

Comment:
is the empty `<>` brackets needed? If not I'd remove them

```suggestion
  std::uniform_int_distribution distrib(1, dCounter.GetValue());
```

Reply:
I think they are necessary to tell the compiler to use the default result type (int); without the `<>` it doesn't compile for me. Mirrors usage in the example in https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution

Reply:
ah thanks, learned something!